diff --git a/third_party/WebKit/Source/core/loader/ProgressTracker.cpp b/third_party/WebKit/Source/core/loader/ProgressTracker.cpp
index 321d176..eb6f206 100644
--- a/third_party/WebKit/Source/core/loader/ProgressTracker.cpp
+++ b/third_party/WebKit/Source/core/loader/ProgressTracker.cpp
@@ -95,7 +95,6 @@ void ProgressTracker::dispose()
 {
     if (m_inProgress)
         progressCompleted();
-    ASSERT(!m_frame->isLoading());
 }
 
 double ProgressTracker::estimatedProgress() const
