diff --git a/components/storage_monitor/storage_monitor_linux.cc b/components/storage_monitor/storage_monitor_linux.cc
index a68e3e8..8cb6ed2 100644
--- a/components/storage_monitor/storage_monitor_linux.cc
+++ b/components/storage_monitor/storage_monitor_linux.cc
@@ -9,6 +9,7 @@
 #include <mntent.h>
 #include <stdint.h>
 #include <stdio.h>
+#include <sys/stat.h>
 #include <limits>
 #include <list>
 #include <utility>
